cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -313,170 +278,29 @@ func (m *gceManagerImpl) assertGCE() {
-				autoprovisioned: false, // NAP is disabled
-				nodePoolName:    nodePool.Name,
-				minSize:         int(nodePool.Autoscaling.MinNodeCount),
-				maxSize:         int(nodePool.Autoscaling.MaxNodeCount),
-			}
-			existingMigs[mig.GceRef] = struct{}{}
-
-			if m.RegisterMig(mig) {
-				changed = true
-			}
-		}
-	}
-	for _, mig := range m.getMigs() {
-		if _, found := existingMigs[mig.config.GceRef]; !found {
-			m.UnregisterMig(mig.config)
-			changed = true
-		}
-	}
-	if changed {
-		m.cacheMutex.Lock()
-		defer m.cacheMutex.Unlock()
-
-		if err := m.regenerateCache(); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-// Gets all registered node pools
-func (m *gceManagerImpl) fetchAllNodePoolsGkeRegionalImpl() error {
-	m.assertGKE()
-
-	nodePoolsResponse, err := m.gkeBetaService.Projects.Locations.Clusters.NodePools.List(fmt.Sprintf("projects/%s/locations/%s/clusters/%s", m.projectId, m.location, m.clusterName)).Do()
-	if err != nil {
-		return err
-	}
-
-	existingMigs := map[GceRef]struct{}{}
-	changed := false
-
-	for _, nodePool := range nodePoolsResponse.NodePools {
-		autoscaled := nodePool.Autoscaling != nil && nodePool.Autoscaling.Enabled
-		if !autoscaled {
-			continue
-		}
-		for _, igurl := range nodePool.InstanceGroupUrls {
-			project, zone, name, err := ParseIgmUrl(igurl)
-			if err != nil {
-				return err
-			}
-			mig := &Mig{
-				GceRef: GceRef{
-					Name:    name,
-					Zone:    zone,
-					Project: project,
-				},
-				gceManager:      m,
-				exist:           true,
-				autoprovisioned: false, // NAP is disabled
-				nodePoolName:    nodePool.Name,
-				minSize:         int(nodePool.Autoscaling.MinNodeCount),
-				maxSize:         int(nodePool.Autoscaling.MaxNodeCount),
-			}
-			existingMigs[mig.GceRef] = struct{}{}
-
-			if m.RegisterMig(mig) {
-				changed = true
-			}
-		}
-	}
-	for _, mig := range m.getMigs() {
-		if _, found := existingMigs[mig.config.GceRef]; !found {
-			m.UnregisterMig(mig.config)
-			changed = true
-		}
-	}
-	if changed {
-		m.cacheMutex.Lock()
-		defer m.cacheMutex.Unlock()
-
-		if err := m.regenerateCache(); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-// Gets all registered node pools
-func (m *gceManagerImpl) fetchAllNodePoolsGkeNapImpl() error {
-	m.assertGKENAP()
-
-	nodePoolsResponse, err := m.gkeBetaService.Projects.Zones.Clusters.NodePools.List(m.projectId, m.location, m.clusterName).Do()
+	nodePools, err := m.GkeService.FetchNodePools()
 	if err != nil {
 		return err
 	}
 
 	existingMigs := map[GceRef]struct{}{}
 	changed := false
 
-	for _, nodePool := range nodePoolsResponse.NodePools {
-		autoprovisioned := nodePool.Autoscaling != nil && nodePool.Autoscaling.Autoprovisioned
-		autoscaled := nodePool.Autoscaling != nil && nodePool.Autoscaling.Enabled
-		if !autoscaled {
-			if autoprovisioned {
-				glog.Warningf("NodePool %v has invalid config - autoprovisioned, but not autoscaled. Ignoring this NodePool.", nodePool.Name)
-			}
-			continue
```

Comment:
What's the reason for removing this logic? I'm not sure it was correct to start with, but I'm trying to understand if it's a conscious decision? You could still do the same check using NodePool struct.

Reply:
Looking at it more closely - you don't need this if you filter in FetchNodePools(). But in FetchNodePools() you only check if nodePool.Autoscaling != nil, not if nodePool.Autoscaling.Enabled is true.

Reply:
Done - added checking Autoscaling.Enabled in FetchNodePools